app/server/src/main/scala/org/bitcoins/server/BitcoindRpcBackendUtil.scala

```diff
@@ -16,6 +16,38 @@ import scala.concurrent.{ExecutionContext, Future, Promise}
 /** Useful utilities to use in the wallet project for syncing things against bitcoind */
 object BitcoindRpcBackendUtil extends BitcoinSLogger {
 
+  def catchupWalletToBitcoind(bitcoind: BitcoindRpcClient, wallet: Wallet)(
+      implicit ec: ExecutionContext): Future[Unit] = {
+    for {
+      bitcoindHeight <- bitcoind.getBlockCount
+      walletStateOpt <- wallet.getSyncHeight()
+      _ <- walletStateOpt match {
+        case None => FutureUtil.unit // New wallet, nothing to sync
+        case Some(syncHeight) =>
+          if (syncHeight.height > bitcoindHeight) {
+            Future.failed(
+              new RuntimeException("Bitcoind is not caught up to wallet"))
```

Comment:
This is a little bit more troubling than bitcoind not being caught up? For instance, are you on the right network? Maybe there was a huge reorg (testnet?). I don't think this case is as simple as "we are ahead of bitcoind"